src/hw_isolation_record/manager.cpp

```diff
@@ -521,6 +521,24 @@ void Manager::updateEntryForRecord(const openpower_guard::GuardRecord& record,
     {
         entryIt->second->resolveEntry(false);
     }
+    else if ((record.recordId != 0xFFFFFFFF) && (entryIt->second->resolved()))
```

Comment:
At line 520 they entry is not resolved so why arewe setting it to false again? not related to your changes but curious.

Reply:
At line 367 if recordId == 0xFFFFFFFF you are assuming it as resolved whey are we checking with != here?

Reply:
OK, there are two possibilities in the GUARD partition.

- Host might be resolved the existing unresolved record so, we need to update that in the respective D-Bus entry. (LN 520, FYI, false is [referring](https://github.com/ibm-openbmc/openpower-hw-isolation/blob/main/include/hw_isolation_record/entry.hpp#L86) to something else not for the resolve property value).
- Host might be overwritten the existing resolved record slot for a new unresolved record so, we need to update that in the respective D-Bus entry (LN 524). 

Note that other details are updated separately based on the value (LN 510, 515)

**Background:**

The host application writing a new record/updating an existing record in their persisted file (aka GUARD partition) so, we need to refresh that into the D-Bus that can be used in the Redfish to list out the record to the end-user.